### PR TITLE
Resolve merge conflicts in handwriting_recognition_service.py and integrate changes from main and feature/round-trip

### PR DESCRIPTION
**Pull request by rmulligan**
## Summary by Sourcery

Refactor and enhance the handwriting recognition service to integrate MyScript iink SDK with improved authentication, error handling, and round-trip workflow support

New Features:
- Add round-trip workflow for processing handwritten queries
- Implement CLI command for round-trip processing
- Create comprehensive test suite for handwriting recognition

Enhancements:
- Improve MyScript iink SDK integration with proper authentication
- Add robust error handling in handwriting recognition service
- Enhance configuration management with environment variable support
- Implement more detailed logging

Documentation:
- Add documentation for round-trip workflow
- Document configuration and usage of new features

Tests:
- Add comprehensive test coverage for handwriting recognition service
- Create test suite for round-trip workflow
- Add error handling and edge case tests

Chores:
- Refactor configuration management
- Update project structure to support new services

### CODE REVIEW

File: tests/test_round_trip.py [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+
+
+@pytest.fixture
+def round_trip_service(mock_handwriting_service, mock_document_service, mock_remarkable_service):
+    """Create a RoundTripService with mock dependencies."""
+    return RoundTripService(
+        handwriting_service=mock_handwriting_service,
+        document_service=mock_document_service,
+        remarkable_service=mock_remarkable_service
+    )
```
[the remark this answers is not in the document]
Comment (rmulligan) [reply tests/test_round_trip.py:88]: @sourcery-ai create issue

File: tests/test_handwriting_recognition.py [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+    )
+    
+    test_data = {"test": "data"}
+    headers = service._generate_headers(test_data)
+    
+    assert "Accept" in headers
+    assert "Content-Type" in headers
+    assert "applicationKey" in headers
+    assert "hmacSignature" in headers
+    assert "hmacTimestamp" in headers
```
[the remark this answers is not in the document]
Comment (rmulligan) [reply tests/test_handwriting_recognition.py:157]: @sourcery-ai create issue

File: tests/test_handwriting_recognition.py [hunk 3]
```diff
@@ -0,0 +1,158 @@
+"""Tests for the handwriting recognition service."""
+
+import os
+import pytest
+from unittest.mock import MagicMock, patch
+
+from inklink.services.handwriting_recognition_service import HandwritingRecognitionService
```
[the remark this answers is not in the document]
Comment (rmulligan) [reply tests/test_handwriting_recognition.py:7]: @sourcery-ai create issue

File: tests/test_handwriting_recognition.py [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+
+
+def test_init_with_keys():
+    """Test initialization with explicit keys."""
+    service = HandwritingRecognitionService(
+        application_key="test_app_key",
+        hmac_key="test_hmac_key"
+    )
+    assert service.application_key == "test_app_key"
+    assert service.hmac_key == "test_hmac_key"
```
[the remark this answers is not in the document]
Comment (rmulligan) [reply tests/test_handwriting_recognition.py:31]: @sourcery-ai create issue

File: tests/test_handwriting_recognition.py [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+
+
+def test_generate_headers():
+    """Test HMAC signature generation."""
+    service = HandwritingRecognitionService(
+        application_key="test_app_key",
+        hmac_key="test_hmac_key"
+    )
+    
+    test_data = {"test": "data"}
```
[the remark this answers is not in the document]
Comment (rmulligan) [reply tests/test_handwriting_recognition.py:150]: @sourcery-ai create issue

File: tests/test_handwriting_recognition.py [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+def test_extract_strokes_empty_file(tmp_path):
+    """Test extracting strokes from an empty file."""
+    service = HandwritingRecognitionService(
+        application_key="test_app_key",
+        hmac_key="test_hmac_key"
+    )
+    
+    # Create an empty file that is not a valid .rm file
+    empty_file = tmp_path / "empty.rm"
+    with open(empty_file, "wb") as f:
```
[the remark this answers is not in the document]
Comment (rmulligan) [reply tests/test_handwriting_recognition.py:73]: @sourcery-ai create issue

File: tests/test_handwriting_recognition.py [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+def test_convert_to_iink_format(mock_strokes):
+    """Test conversion to iink format."""
+    service = HandwritingRecognitionService(
+        application_key="test_app_key",
+        hmac_key="test_hmac_key"
+    )
+    
+    result = service.convert_to_iink_format(mock_strokes)
+    
+    assert result["type"] == "inkData"
```
[the remark this answers is not in the document]
Comment (rmulligan) [reply tests/test_handwriting_recognition.py:53]: @sourcery-ai create issue

File: src/inklink/services/handwriting_recognition_service.py [hunk 8]
```diff
@@ -25,38 +1,51 @@
...
+from inklink.services.interfaces import IHandwritingRecognitionService
+from inklink.utils import retry_operation, format_error
+from inklink.config import CONFIG
 
-    def export(self, content_id: str, format_type: str = "text") -> Dict[str, Any]:
-        # Placeholder: Replace with actual export logic
-        return {"content_id": content_id, "format": format_type, "data": "exported content"}
+logger = logging.getLogger(__name__)
 
 class HandwritingRecognitionService(IHandwritingRecognitionService):
```
[the remark this answers is not in the document]
Comment (rmulligan) [reply src/inklink/services/handwriting_recognition_service.py:23]: @sourcery-ai create issue

File: src/inklink/main.py [hunk 9]
```diff
@@ -31,6 +31,34 @@
...
+@click.option("--output", "-o", help="Output file name (optional)")
+def roundtrip(input_file, output):
+    """Process a handwritten query and generate a response."""
+    from inklink.services.round_trip_service import RoundTripService
+    
+    service = RoundTripService()
+    success, result = service.process_handwritten_query(input_file)
+    
+    if success:
+        click.echo(f"Round-trip processing successful!")
```
[the remark this answers is not in the document]
Comment (rmulligan) [reply src/inklink/main.py:45]: @sourcery-ai create issue

File: src/inklink/services/handwriting_recognition_service.py [hunk 10]
```diff
@@ -66,54 +55,138 @@
...
         try:
-            # Placeholder: Convert strokes to MyScript/iink format
-            # This should be replaced with actual conversion logic
-            return {"strokes": strokes}
+            iink_data = {
+                "type": "inkData",
+                "width": CONFIG.get("PAGE_WIDTH", 1872),
+                "height": CONFIG.get("PAGE_HEIGHT", 2404),
+                "strokes": strokes
+            }
```
[the remark this answers is not in the document]
Comment (rmulligan) [reply src/inklink/services/handwriting_recognition_service.py:123]: @sourcery-ai create issue

File: src/inklink/services/handwriting_recognition_service.py [hunk 11]
```diff
@@ -66,54 +55,138 @@
...
-        self,
-        iink_data: Dict[str, Any],
-        content_type: str = "Text",
-        language: str = "en_US"
-    ) -> Dict[str, Any]:
+    def recognize_handwriting(self, iink_data: Dict[str, Any], content_type: str = "Text", language: str = "en_US") -> Dict[str, Any]:
+        """Process ink data through the iink SDK and return recognition results."""
         try:
-            return self.myscript.recognize(iink_data, content_type, language)
+            if not self.application_key or not self.hmac_key:
```
[the remark this answers is not in the document]
Comment (rmulligan) [reply src/inklink/services/handwriting_recognition_service.py:132]: @sourcery-ai create issue

File: src/inklink/services/handwriting_recognition_service.py [hunk 12]
```diff
@@ -66,54 +55,138 @@
...
-            return {"error": str(e)}
+            error_msg = format_error("recognition", "Handwriting recognition failed", e)
+            logger.error(error_msg)
+            return {"success": False, "error": str(e)}
 
     def export_content(self, content_id: str, format_type: str = "text") -> Dict[str, Any]:
+        """Export recognized content in the specified format."""
         try:
-            return self.myscript.export(content_id, format_type)
+            if not self.application_key or not self.hmac_key:
```
[the remark this answers is not in the document]
Comment (rmulligan) [reply src/inklink/services/handwriting_recognition_service.py:173]: @sourcery-ai create issue
